### ISSUE TITLE
[Accessibility]: va-button-segmented has insufficient color contrast

### ISSUE THREAD

**Issue by bacitracin** [department-of-veterans-affairs/vets-design-system-documentation#5894]
### Point of contact

@bacitracin

### How severe do you think this issue is?

Medium

### Severity rationale

Contrast could be better, but it's still usable.

### URL where issue occurs

https://design.va.gov/storybook/?path=/docs/va-button-segmented--docs

### Component or pattern (if applicable)

va-button-segmented

### Environment details

OS: Mac
Browser: Chrome

### Steps to reproduce

Came up in accessibility testing for staging review of the Claim Status tools' Finding Claims feature, which uses the segmented button. This is reproducible on Storybook https://design.va.gov/storybook/?path=/docs/va-button-segmented--docs. Used the Color Contrast Analyzer app.

The light grey of the segmented button against white (background and active button) has a contrast ratio of 1.3:1, below the required 3:1 to meet WCAG 1.4.11.

The light grey against the dark grey is also 1.3:1.

<img width="1079" height="679" alt="Image" src="https://github.com/user-attachments/assets/3b878ea2-7f24-4338-a5bd-14710d470eea" />
<img width="1090" height="684" alt="Image" src="https://github.com/user-attachments/assets/eca48641-c5e8-4977-8b21-232774ea6a7e" />

...

### Type of issue

- [ ] Axe-core
- [ ] Cognitive | user flows & navigating
- [x] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [ ] Focus
- [ ] Headings
- [ ] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Screenreader | assistive tech & device support
- [ ] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else
